MSVC build is broken, no countl_zero and popcount

With Visual Studio 2022 I get these errors when building zxing-cpp 2.1.0:

```
  zxing-cpp\core\src\BitHacks.h(41,14): error C2039: 'countl_zero': is not a member of 'std' [zxing-cpp\build\temp.win32-cpython-38\Release\zxingcpp.vcxproj]
  C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.37.32822\include\vector(26,1): message : see declaration of 'std' [zxing-cpp\build\temp.win32-cpython-38\Release\zxingcpp.vcxproj]
  zxing-cpp\core\src\BitHacks.h(84,14): error C2039: 'countr_zero': is not a member of 'std' [zxing-cpp\build\temp.win32-cpython-38\Release\zxingcpp.vcxproj]
  C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.37.32822\include\vector(26,1): message : see declaration of 'std' [zxing-cpp\build\temp.win32-cpython-38\Release\zxingcpp.vcxproj]
  zxing-cpp\core\src\BitHacks.h(153,14): error C2039: 'popcount': is not a member of 'std' [zxing-cpp\build\temp.win32-cpython-38\Release\zxingcpp.vcxproj]
  C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.37.32822\include\vector(26,1): message : see declaration of 'std' [zxing-cpp\build\temp.win32-cpython-38\Release\zxingcpp.vcxproj]
  zxing-cpp\core\src\BitHacks.h(153,14): error C3861: 'popcount': identifier not found [zxing-cpp\build\temp.win32-cpython-38\Release\zxingcpp.vcxproj]
```

This seems to be caused by https://github.com/zxing-cpp/zxing-cpp/commit/9980b2da410c4194578d942cd1d6d4d619a05722 and building version 2.0.0 works.

--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -22,6 +22,11 @@ if (WINRT)
         -DWINRT
     )
 endif()
+if (MSVC)
+    set (ZXING_CORE_DEFINES ${ZXING_CORE_DEFINES}
+        /Zc:__cplusplus
+    )
+endif()
 
 set (ZXING_CORE_LOCAL_DEFINES
     $<$<BOOL:${BUILD_READERS}>:-DZXING_BUILD_READERS>
@@ -34,7 +39,6 @@ if (MSVC)
         -D_CRT_SECURE_NO_WARNINGS
         -D_CRT_NONSTDC_NO_WARNINGS
         -DNOMINMAX
-        /Zc:__cplusplus
     )
 else()
     set (ZXING_CORE_LOCAL_DEFINES ${ZXING_CORE_LOCAL_DEFINES}

